src/Core/Result.php

```diff
@@ -115,31 +115,4 @@ public function __construct(ResponseInterface $response, HttpClientInterface $ht
     protected function populateResult(ResponseInterface $response, ?HttpClientInterface $httpClient): void
     {
     }
-
-    final protected function xmlValueOrNull(?\SimpleXMLElement $xml, string $type)
-    {
-        if (null === $xml) {
-            return null;
-        }
-
-        $value = $xml->__toString();
-        if (empty($value) && 0 === $xml->count()) {
```

Comment:
I removed the `if (count($xlk) === 0) return null`, because I'm not sure this is true.
In official php sdk, this is not the case. for instance, in S3::ListObjects => the response contains `<Prefix></Prefix>` and php returns an empty string `""` see https://docs.aws.amazon.com/aws-sdk-php/v3/api/api-s3-2006-03-01.html#listobjectsv2-example-1

Reply:
Note that `$xml->foobar` will always return an SimpleXmlElement. No matter how the XML looks like. That is why we check if "foobar" exists. We do that with `$xml->foobar->count() === 0`

Reply:
yes but casted to book it returns false.

```
$xml->exist ? 'true' : 'false'; // true
$xml->doesNot ? 'teue' : 'false'; // false
```

I'm AFK, but I'll provide a 3v4l + open an issue on phpstan for that

Reply:
Added a test to cover that case

Reply:
Bug opened on phpstan (to remove the rule) see https://github.com/phpstan/phpstan/issues/2997